File: app/src/main/java/com/example/pam_project/database/relationships/ListsWithTasks.java
package com.example.pam_project.database.relationships;

import androidx.room.Embedded;
import androidx.room.Relation;

import com.example.pam_project.database.lists.ListEntity;
import com.example.pam_project.database.tasks.TaskEntity;

import java.util.List;

public class ListsWithTasks {
    @Embedded
    public ListEntity list;
    @Relation(
            parentColumn = "id",
            entityColumn = "listId"
    )
    public List<TaskEntity> tasks;
}

File: app/src/main/java/com/example/pam_project/database/relationships/ListsWithTasks.kt
package com.example.pam_project.database.relationships

import androidx.room.Embedded
import androidx.room.Relation
import com.example.pam_project.database.lists.ListEntity
import com.example.pam_project.database.tasks.TaskEntity

data class ListsWithTasks (@Embedded var list: ListEntity,
                           @Relation(parentColumn = "id", entityColumn = "listId") var tasks: List<TaskEntity>)